aries_vcx/src/libindy/utils/ledger.rs

```diff
@@ -352,6 +354,40 @@ pub async fn get_service(pool_handle: PoolHandle, did: &Did) -> VcxResult<AriesS
     })
 }
 
+pub async fn into_did_doc(pool_handle: PoolHandle, invitation: &Invitation) -> VcxResult<DidDoc> {
```

Comment:
I think this actually makes sense to be under `messages` module rather, however instead of 
```
           let service = get_service(pool_handle, &invitation.did).await.unwrap_or_else(|err| {
                error!("Failed to obtain service definition from the ledger: {}", err);
                AriesService::default()
            });
            (service.service_endpoint, service.recipient_keys, service.routing_keys)
```
we should imo have something like 
```
        Invitation::Public(invitation) => {
            did_doc.set_id(invitation.did.to_string());
            return indy_resolver::resolve_ddo(invitation.did);
        }
```
And ddo_resolver could be trait, I imagine we could have our own ddo resolution, or we could also implement that trait experimentally by calling http://uniresolver.io/

Reply:
Doing this, `into_did_doc` could then stay in messages

Reply:
But creating ddo resolver interface and start using it in the codebase might be out of scope here. So just a suggestion for a next steps :-)

Reply:
I agree, but my goal here is just to remove all mentions of handle as simply as possible and then move code back in when it makes sense.